default/AI/AIstate.py

```diff
@@ -662,6 +661,18 @@ def rate_fleet(self, fleet_id, enemy_stats=None, ship_info=None):
                 shields = ship.currentMeterValue(fo.meterType.shield)
                 stats['structure'] = structure
                 stats['shields'] = shields
+                stats['attack'] = 0
+                stats['attacks'] = {}
+                design = ship.design
+                if not design:
+                    print "ERROR: Ship has no valid design :)"
+                    print "If this happened in a moderated game: Screw those. The AI simply does not work with them."
+                    continue
+                for partname in design.parts:
+                    if partname and fo.getPartType(partname).partClass == fo.shipPartClass.shortRange:
+                        damage = ship.currentPartMeterValue(fo.meterType.capacity, partname)
+                        stats['attack'] += damage
+                        stats['attacks'][damage] = stats['attacks'].get(damage, 0) + 1
```

Comment:
I do not redefine it, I reset its values as after ```get_weighted_design_stats(design_id, species_name)``` they contain the corresponding stats of the design.

Probably, we could ignore the design completely and only loop over ship stats but that is cleanup beyond the scope of this fix.
Of course, at some point in time, we also want to adress that part of the code but add it onto the list of 100 parts of the AI code which need to be cleaned up, reformatted, documented and so on...

I am busy rewriting other parts of the code atm, so I won't bother with cleaning up this part. Writing 2 clean lines in an environment of 1000 messy ones is a waste of time if it needs to think about longer than 10 seconds... Better do it all at once.
As long as the bugfix works, that should be all required to merge this PR. If you want to clean the mess up, feel free to do so, but probably should be done in a later PR as this is a very relevant bug.